lib/spack/spack/binary_distribution.py

```diff
@@ -486,9 +485,15 @@ class PickKeyException(spack.error.SpackError):
     """
 
     def __init__(self, keys):
-        err_msg = "Multiple keys available for signing\n%s\n" % keys
-        err_msg += "Use spack buildcache create -k <key hash> to pick a key."
-        super(PickKeyException, self).__init__(err_msg)
+        msg = "Multiple keys available for signing\n---"
+        for i, k in enumerate(keys):
+            msg += "\n"
+            msg += "{0: <3} User ID: {1}\n".format("{0}.".format(i + 1), k.uid)
+            msg += "{0: <3} Fingerprint: {1}\n".format("", k.fingerprint)
+        msg += "---\n"
+        msg += "Specify your key choice unambiguously using `--key <ID>`. "
+        msg += "<ID> can be the fingerprint or user-id.\n"
```

Comment:
Woohoo!  I wasn't aware we could use the ID when I first encountered the multiple key error.